src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentTransactionInfoPlugin.java

```diff
@@ -315,6 +348,38 @@ private static PaymentPluginStatus pspResultToPaymentPluginStatus(final PaymentS
         }
     }
 
+    private static List<PluginProperty> extractPluginProperties(PurchaseResult purchaseResult) {
+        ImmutableMap.Builder<String, String> builder = ImmutableMap.builder();
+        if (purchaseResult.getFormParameter() != null) {
+            builder.putAll(purchaseResult.getFormParameter());
+        }
+
+        Map<String, String> additionalData = purchaseResult.getAdditionalData();
+
+        if (additionalData != null) {
+            String avsResultRaw = additionalData.get("avsResultRaw");
+            String avsResult = additionalData.get("avsResult");
+            String cvcResultRaw = additionalData.get("cvcResultRaw");
+            String cvcResult = additionalData.get("cvcResult");
+
+            if (avsResultRaw != null) {
+                //use Unknown (UK) as default
+                builder.put("avsResultCode", CONVERT_AVS_CODE.getOrDefault(avsResultRaw, "UK"));
```

Comment:
I would also return `avsResultRaw`, as it contains more information.

Reply:
`avsResultRaw` contains just the code, `avsResult` contains the code and the text message.

Reply:
For example:
```
<entry>
  <key xsi:type="xsd:string">avsResult</key>
  <value xsi:type="xsd:string">4 AVS not supported for this card type</value>
</entry>
<entry>
  <key xsi:type="xsd:string">avsResultRaw</key>
  <value xsi:type="xsd:string">4</value>
</entry>
```